chroma_core/plugins/block_devices.py

```diff
@@ -12,23 +12,51 @@
 log.setLevel(DEBUG)
 
 
-def _fetch_aggregator():
+def _fetch_aggregator(timeout):
     import requests
 
-    resp = requests.get(settings.DEVICE_AGGREGATOR_URL)
-    payload = resp.text
-
-    return json.loads(payload)
-
-
-def get_devices(fqdn):
-    try:
-        _data = _fetch_aggregator()
-        return _data[fqdn]
-    except Exception as e:
-        log.error(
-            "iml-device-aggregator is not providing expected data, ensure "
-            "iml-device-scanner package is installed and relevant "
-            "services are running on storage servers (%s)" % e
-        )
-        return {}
+    summary = 0.0
+    timeout = timeout if timeout >= 0 else 0
+    while summary <= timeout:
+        resp = requests.get(settings.DEVICE_AGGREGATOR_URL)
+        summary += resp.elapsed.seconds
+        summary += resp.elapsed.microseconds / 10.0 ** 6
+
+        try:
+            return resp.json()
```

Comment:
This variant is not returning a tuple, which is causing testing to fail.

```suggestion
            return resp.json(), summary
```